presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/RemoveNullRowInAggregation.java

```diff
@@ -0,0 +1,105 @@
+import com.facebook.presto.spi.relation.SpecialFormExpression;
+import com.facebook.presto.sql.planner.iterative.Rule;
+import com.facebook.presto.sql.relational.FunctionResolution;
+import com.google.common.collect.ImmutableList;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+import static com.facebook.presto.common.type.BooleanType.BOOLEAN;
+import static com.facebook.presto.metadata.FunctionAndTypeManager.createTestFunctionAndTypeManager;
+import static com.facebook.presto.spi.plan.AggregationNode.Aggregation;
+import static com.facebook.presto.spi.relation.SpecialFormExpression.Form.AND;
+import static com.facebook.presto.spi.relation.SpecialFormExpression.Form.IS_NULL;
+import static com.facebook.presto.sql.planner.plan.Patterns.aggregation;
+
+public class RemoveNullRowInAggregation
+        implements Rule<AggregationNode>
+{
+    private static final Pattern<AggregationNode> PATTERN = aggregation()
+            .matching(RemoveNullRowInAggregation::checkArgumentNum);
+
+    private static boolean checkArgumentNum(AggregationNode node)
+    {
+        PlanNode source = node.getSource();
+        if (source instanceof FilterNode) {
+            return false;
+        }
+        Collection<Aggregation> aggregations = node.getAggregations().values();
+        for (Aggregation aggregation : aggregations) {
+            if (aggregation.getCall().getArguments().size() > 1) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    @Override
+    public Pattern<AggregationNode> getPattern()
+    {
+        return PATTERN;
+    }
+
+    @Override
+    public Result apply(AggregationNode aggregationNode, Captures captures, Context context)
+    {
+        PlanNode source = aggregationNode.getSource();
+        if (context.getLookup().resolveGroup(source).findFirst().filter(node -> node instanceof FilterNode)
+                .isPresent()) {
+            return Result.empty();
+        }
+        Collection<Aggregation> aggregations = aggregationNode.getAggregations().values();
+
+        //extract all the aggregation arguments
+        List<RowExpression> rowExpressionList = new ArrayList<>();
+        for (Aggregation aggregation : aggregations) {
+            rowExpressionList.add(aggregation.getCall().getArguments().get(0));
+        }
+
+        //Form all the ISNULL form
+        List<RowExpression> nullFormList = new ArrayList<>();
+        for (RowExpression rowExpression : rowExpressionList) {
+            RowExpression nullForm = new SpecialFormExpression(IS_NULL, BOOLEAN, rowExpression);
+            nullFormList.add(nullForm);
+        }
+
+        RowExpression andForm = new SpecialFormExpression(AND, BOOLEAN, ImmutableList.copyOf(nullFormList));
+        RowExpression notCallPredicate = new CallExpression("not",
+                new FunctionResolution(createTestFunctionAndTypeManager().getFunctionAndTypeResolver()).notFunction(),
```

Comment:
Do not use createTestFunctionAndTypeManager. Get it from the optimizer metadata. See other optimizer [rules](https://github.com/jackychen718/presto/blob/Drop-Agg-NULL/presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/SimplifyCountOverConstant.java#L55) for examples

Reply:
ok